core/file_walker.go

```diff
@@ -1,50 +1,70 @@
 package core
 
 import (
-	"fmt"
+	"log"
 	"os"
 	"path/filepath"
 )
 
 type File struct {
 	Name  string
+	Dir   *File
```

Comment:
That's a shame, this message had to get lost in the sheer amount of our conversation :) https://github.com/viktomas/godu/pull/18#discussion_r95131458

`State.ancestors` is used in `GoBack` command. It is different solution for the exact same problem (how to get from subdir to a parent dir)

My reasoning was that the file tree is going to have milions of files and every pointer/field I won't put into `File` struct is going to save a lot of space. When I compared memory footprint of godu with and without `Dir` in the `File` struct it increased (on my harddrive) from 180MB -> 203MB of memory.